### DISCUSSION TITLE
is there a flag equivalent to jest "changedSince" flag?

### DISCUSSION THREAD

**Discussion by Madvinking** [vitest-dev/vitest#1688] [Q&A]
in our CI pipeline, i would like to run only tests related to the file changed from my last commit

**Comment by Demivan** [comment]
Yes. There is `--changed` flag.

https://vitest.dev/guide/cli.html#changed

**Reply by akashdeep-sharechat** [reply]
@Demivan  I have tried running this with branch, which the docs suggest is possible, but am getting this error

`Error: fatal: ambiguous argument 'origin/master...HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
 ❯ makeError node_modules/vitest/dist/chunk-vite-node-externalize.58e10976.mjs:9695:11
 ❯ handlePromise node_modules/vitest/dist/chunk-vite-node-externalize.58e10976.mjs:10102:26
 ❯ process.processTicksAndRejections node:internal/process/task_queues:95:5
 ❯ VitestGit.resolveFilesWithGitCommand node_modules/vitest/dist/chunk-vite-node-externalize.58e10976.mjs:10154:16
 ❯ VitestGit.findChangedFiles node_modules/vitest/dist/chunk-vite-node-externalize.58e10976.mjs:10168:47
 ❯ Vitest.filterTestsBySource node_modules/vitest/dist/chunk-vite-node-externalize.58e10976.mjs:10333:24
 ❯ Vitest.start node_modules/vitest/dist/chunk-vite-node-externalize.58e10976.mjs:10289:19
 ❯ startVitest node_modules/vitest/dist/chunk-vite-node-externalize.58e10976.mjs:11073:5
 ❯ CAC.start node_modules/vitest/dist/cli.mjs:666:9

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: {
  "command": "git diff --name-only origin/master...HEAD",
  "escapedCommand": "git diff --name-only \"origin/master...HEAD\"",
  "exitCode": 128,
  "failed": true,
  "isCanceled": false,
  "killed": false,
  "shortMessage": "Command failed with exit code 128: git diff --name-only origin/master...HEAD",
  "signal": undefined,
  "signalDescription": undefined,
  "stderr": "fatal: ambiguous argument 'origin/master...HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'",
  "stdout": "",
  "timedOut": false,
}`

**Reply by Demivan** [reply]
Do you actually have `origin/master`? I get this error if branch does not exist

**Reply by akashdeep-sharechat** [reply]
Yes yes I have , the full command was "vitest --changed origin/master"

**Reply by Demivan** [reply]
What is the output of git when running manually?

`git diff --name-only "origin/master...HEAD"`
and
`git log origin/master` - do you get log messages?
